homework-submission/week2/my-app/src/components/ToDoItems.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react"
+import Post from "../Post"
+
+import ToDos from "../Todos"
+
+class TodoItem extends Component {
+   constructor(props) {
+      super(props)
+      this.state = {
+         Todos: ToDos
+      }
+      this.onChange = this.onChange.bind(this)
+   }
+
+   onChange(id) {
+      this.setState(prevState => {
+         const updatedTodos = prevState.Todos.map(todo => {
+            if (todo.id === id) {
+               todo.completed = !todo.completed
+            }
+            return todo
+         })
+         return {
+            Todos: updatedTodos
+         }
+      })
```

Comment:
The loop here is costly, because you are looping over all the data to find one item which you want to change.
and you have in line 35 a key in the Map loop, so you can pass this key as a prop to the < Post > component and use it to find the item